doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+
+            A comprehensive list of possible errors:
+            - `ENTRY_DOES_NOT_EXIST` if an entry with the given key (path) does not exist in the database
+
+
+## Create database entry [POST]
+
+Can be used to create a new entry in the database. An entry consists of a unique key (path) that is built from several individual inputs, a title, description and a configuration snippet in any supported format.
+
++ Request (application/json)
+    + Attributes (object)
+        + organization: `org.apache` (string, required)
+
+            The organization which built the software for which the configuration is meant.
+
+            Regex: ^[a-zA-Z0-9\.\-]+$
+
+        + application: `hive` (string, required)
+
+            The application for which the configuration is meant.
+
+            Regex: ^[a-zA-Z0-9\.\-]+$
+
+        + scope: `database` (string, required)
+
+            Often different parts of an application have different configurations. Here the application specific scope can be defined.
+
+            Regex: ^[a-zA-Z0-9\.\-]+$
+
+        + slug: `sample-snippet-for-devs` (string, required)
+
+            A unique slug that identifies the entry and configuration snippet further.
+
+            Regex: ^[a-zA-Z0-9\.\-]+$
+
+        + title: `Sample snippet for developers` (string, required)
+
+            A title for the entry that ideally explains already what the snippet is about.
+
+            Regex: ^[^\n\r]{3,}$
+
+        + description: `This is a sample conf snippet for Apaches Hive database module.` (string, required)
+
+            An extended description for the entry that explains in detail what the snippet is about. It can also contain instructions on how to use the snippet.
+
+            Regex: ^[\S\s]{3,}$
+
+        + tags: `dev`, `developer`, `db-dev` (array[string], optional)
+
+            A list of tags that shall be added to the entry. Tags may be used for search, so they ideally add additional information that is not already available by the other inputs.
+
+            Regex: ^[a-z0-9\-\.]{3,20}$
+
+        + configuration: `sampleKey = sampleValue` (string, required)
+        
+            The configuration for which a new database entry should be created. It can be of any supported format, in this example **ini** is used.
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible messages:
+            - `ENTRY_CREATED_SUCCESSFULLY` if the entry has been added successfully to the database
+
++ Response 400 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
```

Comment:
Is this information even necessary for the user? In my eyes he only cares about whether it fails or not, because if it is an internal error he cannot influence, then the information is of no value for him...